src/System.Security.SecureString/tests/SecureStringTests.cs

```diff
@@ -0,0 +1,384 @@
+        [InlineData(1000, false)]
+        [InlineData(1000, true)]
+        public static void SecureStringMarshal_Ansi_Roundtrip(int length, bool allocHGlobal)
+        {
+            string input = new string(Enumerable.Range(0, length).Select(i => (char)('a' + i)).ToArray());
+
+            IntPtr marshaledString = Marshal.StringToHGlobalAnsi(input);
+            string expectedAnsi = Marshal.PtrToStringAnsi(marshaledString);
+            Marshal.FreeHGlobal(marshaledString);
+
+            using (SecureString ss = CreateSecureString(input))
+            {
+                IntPtr marshaledSecureString = allocHGlobal ? 
+                    SecureStringMarshal.SecureStringToGlobalAllocAnsi(ss) :
+                    SecureStringMarshal.SecureStringToCoTaskMemAnsi(ss);
+
+                string actualAnsi = Marshal.PtrToStringAnsi(marshaledSecureString);
+
+                if (allocHGlobal)
+                {
+                    Marshal.FreeHGlobal(marshaledSecureString);
+                }
+                else
+                {
+                    Marshal.FreeCoTaskMem(marshaledSecureString);
+                }
+
+                Assert.Equal(expectedAnsi, actualAnsi);
+            }
+        }
+
+        [Theory]
+        [InlineData(0, false)]
+        [InlineData(0, true)]
+        [InlineData(1, false)]
+        [InlineData(1, true)]
+        [InlineData(2, false)]
+        [InlineData(2, true)]
+        [InlineData(1000, false)]
+        [InlineData(1000, true)]
+        public static void SecureStringMarshal_Unicode_Roundtrip(int length, bool allocHGlobal)
+        {
+            string input = new string(Enumerable.Range(0, length).Select(i => (char)('a' + i)).ToArray());
+
+            IntPtr marshaledString = Marshal.StringToHGlobalUni(input);
+            string expectedAnsi = Marshal.PtrToStringUni(marshaledString);
+            Marshal.FreeHGlobal(marshaledString);
+
+            using (SecureString ss = CreateSecureString(input))
+            {
+                IntPtr marshaledSecureString = allocHGlobal ?
+                    SecureStringMarshal.SecureStringToGlobalAllocUnicode(ss) :
+                    SecureStringMarshal.SecureStringToCoTaskMemUnicode(ss);
+
+                string actualAnsi = Marshal.PtrToStringUni(marshaledSecureString);
+
+                if (allocHGlobal)
+                {
+                    Marshal.FreeHGlobal(marshaledSecureString);
+                }
+                else
+                {
+                    Marshal.FreeCoTaskMem(marshaledSecureString);
+                }
+
+                Assert.Equal(expectedAnsi, actualAnsi);
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public static void Stress_Growth()
+        {
+            string starting = new string('a', 6000);
+            var sb = new StringBuilder(starting);
+            using (SecureString testString = CreateSecureString(starting))
+            {
+                for (int i = 0; i < 4000; i++)
```

Comment:
8,002,000 encrypt+decrypt operations, eh?  (Plus a lot of memcpy... 32GB worth?).  How long does this take?

I know it's OuterLoop already, I'm just curious :)